command/loadbot/result.go

```diff
@@ -103,18 +139,30 @@ func (lr *LoadbotResult) initDetailedErrors(gen generator.TransactionGenerator)
 	lr.DetailedErrorData.DetailedErrorMap = errMap
 }
 
-func (lr *LoadbotResult) writeBlockData(buffer bytes.Buffer) {
-	buffer.WriteString("\n\n[BLOCK DATA]\n")
+func (lr *LoadbotResult) writeBlockData(buffer *bytes.Buffer, blockType string) {
+	var blockData *TxnBlockData
+
+	switch blockType {
+	case "contract":
+		blockData = &lr.ContractBlockData
+
+		buffer.WriteString("\n\n[CONTRACT BLOCK DATA]\n")
+	default:
+		blockData = &lr.BlockData
```

Comment:
Instead of initializing the fields `ContractBlockData` or `BlockData`, where you either use one or the other, you can make an interface method like `GetBlockData`, where your transfer and contract modes implement appropriate result setting. That way we can avoid this switch entirely

Reply:
Dumped the switch ecf9a2e7239e8fb2d1f3d2d03060f8fc38d97208